### PR TITLE
#14 Add eslint to project

### PR DESCRIPTION
**Pull request by jmeline**
Using the eslint-config-red eslint rules.

I turned off a few of the rules as they seemed a little out there such as `` instead of '' for javascript quotes. If you would like them back, I will happily revert those changes.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,9 @@
   "main": "utility.js",
   "scripts": {
     "test": "ava",
-    "perftest": "node --experimental-modules ./tests/performance/deepCopy.js"
+    "perftest": "node --experimental-modules ./tests/performance/deepCopy.js",
+    "lint-fix": "eslint --ignore-path .gitignore --fix .",
```
Comment (jmeline) [inline package.json:10]: For the folder to look at, I had it look at the current folder. If you want me to specify a folder instead, let me know and I can make that changes.

File: package.json [hunk 2]
```diff
@@ -25,12 +27,32 @@
...
+      "ecmaFeatures": {}
+    },
+    "env": {
+      "es6": true,
+      "browser": true
+    },
+    "rules": {
+        "prefer-destructuring": "off",
+        "quotes": "off",
+        "no-magic-numbers": "off",
```
Comment (jmeline) [inline package.json:44]: magic numbers rule and my deepEqual test file didn't mix well.

Comment (jmeline) [reply package.json:44]: I added variables where it made sense to satisfy this rule.

File: typeCast.js [hunk 3]
```diff
@@ -5,7 +5,7 @@ export {
 };
 
 
-const stringFromArrayBuffer = function (arrayBuffer, encoding = "utf-8") {
+const stringFromArrayBuffer = function (arrayBuffer, encoding = `utf-8`) {
```
Comment (jmeline) [inline typeCast.js:8]: The `` instead of "" seems really weird.

File: package.json (hunk 1 above)
Comment (GrosSacASac) [reply package.json:10]: good enough for now

File: typeCast.js (hunk 3 above)
Comment (GrosSacASac) [reply typeCast.js:8]: I understand, it takes some time to get used to it

**Review by GrosSacASac — APPROVED** [review]

**Comment by GrosSacASac** [comment]
very good

**Pull request merged**